radio/src/strhelpers.cpp

```diff
@@ -580,24 +580,26 @@ char *getSourceString(char *dest, mixsrc_t idx)
     div_t qr = div(idx - MIXSRC_FIRST_LUA, MAX_SCRIPT_OUTPUTS);
     if (qr.quot < MAX_SCRIPTS &&
         qr.rem < scriptInputsOutputs[qr.quot].outputsCount) {
-      *dest++ = CHAR_LUA;
 // temporary string
 #define MAX_CHAR 16
-      char temp[MAX_CHAR];
-      strncpy(temp, g_model.scriptsData[qr.quot].name, MAX_CHAR);
-
+      char temp[MAX_CHAR];     
+      strncpy(temp, g_model.scriptsData[qr.quot].name, LEN_SCRIPT_FILENAME);
       // instance Name is empty : dest = n-ScriptFileName/OutputName
-      if (temp[0] == 0) {
-        snprintf(temp, MAX_CHAR, "%d-%s/%s", qr.quot + 1,
-                 g_model.scriptsData[qr.quot].file,
-                 scriptInputsOutputs[qr.quot].outputs[qr.rem].name);
+      if (temp[0] == '\0') {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation="       
```

Comment:
I would prefer adding `-Wno-format-truncation` rather than this. It seems a bit far fetched to me (see https://stackoverflow.com/questions/51534284/how-to-circumvent-format-truncation-warning-in-gcc/51536663).

Reply:
I tries to be the less intrusive as possible. Therefore I don't fully understand your comment here.

Reply:
This sort of behavior from GCC is quite annoying... had to do a bunch of `% 100u` in some of my own `snprintf` print code simply because updated GCC was moaning about more than two digit limited formatting of integers... :-/ 

I suspect Raphael was meaning just enabling it globally since it's a rather stupid warning from `gcc` most of the time... and  since this is radio code it's probably not... right... to do this sort of code (plus it's messy)... 

```
 int ret = snprintf(dst, sizeof(dst), "%s!", src);
 if (ret >= sizeof(dst)) {
    buf[sizeof(dst)-1] = 0;
}
```    

Is avoiding snprintf and going to std::string a viable option? Alternately, I believe one of the GCC devs suggested this as a workaround

```
  #define snprintf_trunc(dst, size, ...) \
    do {                                 \
      volatile size_t n = size;          \
      snprintf (dst, n, __VA_ARGS__);    \
    } while (0)
```

i.e. function that will ensure it is always truncated, thus the warning should not be shown. 
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89312

Just something to think about

Reply:
We not only have the problem of writing into fixed length buffers with or without termination (MTFCA), but we also have the problem of reading from such may-be terminated buffers, also in `snprintf()` statements (mostly concatenating some MTFCA and string-literals).

I made some other suggestions in #1340 using templates. In templated code the compiler is able to deduce the correct length of the arrays. This eliminates errors where the programmers gives the wrong length of target or source arrays, which is still the case in the proposal above with `snprintf_trunc()` (in my opinion `snprintf_trunc()` does not solve really  the problem as root-cause).